include/aws/io/pkcs11.h

```diff
@@ -0,0 +1,64 @@
+#ifndef AWS_IO_PKCS11_H
+#define AWS_IO_PKCS11_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/io/io.h>
+
+struct aws_allocator;
+
+/**
+ * Handle to a loaded PKCS#11 library.
+ */
+struct aws_pkcs11_lib_handle;
+
+/**
+ * Options for aws_pkcs11_lib_handle_new()
+ */
+struct aws_pkcs11_lib_handle_options {
+    /**
+     * Name of PKCS#11 library file to load.
+     * Set to NULL if your application is compiled with PKCS#11 symbols linked in.
```

Comment:
Will the user be able to create handles to a linked PKCS#11 library and dynamically loaded ones?

This isn't something to solve now, but it's a subtle implementation consideration for these handles.

Reply:
Yeah the idea is to eventually support:
1) dynamically loaded (based on filepath at runtime)
2) dynamically linked (because their program against it at build time)
3) statically linked

I think we'll get 1 & 2 immediately. 3 will take some more work but we can support it someday